src/Razor/test/testapps/AppWithP2PReference/AppWithP2PReference.csproj

```diff
@@ -28,10 +28,10 @@
     <ProjectReference Include="..\ClassLibrary\ClassLibrary.csproj"/>
   </ItemGroup>
 
-  <ItemGroup Condition="'$(BinariesRoot)'==''">
+    <PropertyGroup Condition="'$(BinariesRoot)'==''">
     <!-- In test scenarios $(BinariesRoot) is defined in a generated Directory.Build.props file -->
-    <ProjectReference Include="..\..\Microsoft.AspNetCore.Razor.Test.MvcShim.ClassLib\Microsoft.AspNetCore.Razor.Test.MvcShim.ClassLib.csproj"/>
-  </ItemGroup>
+    <BinariesRoot>$(RepositoryRoot)artifacts\bin\Microsoft.AspNetCore.Razor.Test.MvcShim.ClassLib\$(Configuration)\netstandard2.0\</BinariesRoot>
```

Comment:
Having project references that go away during test runs is problematic since ProjectReferences are stored as distinct nodes. Doing this is slightly worse when building these apps outside of the tests, (you have to build the shims first), but it's the only meaningful way forward.

Reply:
I think this is worth adding as a comment in those csprojs.

Reply:
Ehh, the tests fail almost immediately if you do this. The fact we did something entirely different when doing non-tests build was the weirder behavior.